packages/grafana-ui/src/components/Layout/Stack/Stack.tsx

```diff
@@ -2,18 +2,28 @@ import React from 'react';
 
 import { ThemeSpacingTokens } from '@grafana/data';
 
-import { Direction, Flex } from '../Flex/Flex';
+import { Flex } from '../Flex/Flex';
 import { ResponsiveProp } from '../utils/responsiveness';
 
+import { HorizontalStack } from './HorizontalStack';
+
 interface StackProps {
-  direction?: ResponsiveProp<Direction>;
+  direction?: ResponsiveProp<'horizontal' | 'vertical'>;
   gap?: ResponsiveProp<ThemeSpacingTokens>;
 }
 
-export const Stack = ({ gap = 1, direction = 'column', children }: React.PropsWithChildren<StackProps>) => {
-  return (
-    <Flex gap={gap} direction={direction}>
-      {children}
-    </Flex>
-  );
-};
+export const Stack = React.forwardRef<HTMLDivElement, React.PropsWithChildren<StackProps>>(
+  ({ gap = 1, direction = 'vertical', children }, ref) => {
+    return (
+      <Flex ref={ref} gap={gap} direction={direction === 'vertical' ? 'column' : 'row'} wrap="wrap">
```

Comment:
Hmm yeah I don't think the responsiveness will work this way, you would have to loop through each of the breakpoints and convert them from horizontal/vertical to column/row.
If we did have direction just as column/row it would work 👀

Reply:
Although, I might not see the value of changing directions based on media queries for this component.

Reply:
We currently do this with the `Flex` on the `DashboardEmpty` [page](https://github.com/grafana/grafana/pull/73637/files#diff-fe97c057a393cf9bcdbf30c13d864acb55cbe742b59f70a36a723219fc9c1d63R61) . We probably can achieve the same thing with some sort of Grid though

Reply:
>Although, I might not see the value of changing directions based on media queries for this component.

Just my two cents... I think switching from a column layout for mobile devices to row layout for larger devices can be really useful. I've used it a ton in other projects for vertically stacking things like button groups, form elements, or contents of cards for mobile devices.

Reply:
Okay I changed it ot use responsive props